src/main/java/duke/internal/Parser.java

```diff
@@ -0,0 +1,145 @@
+package duke.internal;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * A parser object can parse a command from a string,
+ * and determine which Command object should be used to execute it.
+ */
+public class Parser {
+    /**
+     * Parses a command from a string.
+     *
+     * @param str the string to parse
+     * @return the Command object to execute
+     */
+    public static Command parse(String str) {
```

Comment:
could this be called parseCommand instead? 
_"Names representing methods must be verbs and written in camelCase."_